### PR TITLE
Cancel build: get build from the current project

### PR DESCRIPTION
**Pull request by stsewd**

### CODE REVIEW

**Comment by agjohnson** [comment]
:+1: this is the fix I was assuming as well. I'll come back to approve tests, but they look good eyeballing them really quick. I'll let CI chug away on those in the meantime.

File: readthedocs/builds/tests/test_views.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        self.build.save()
+        self.client.force_login(self.user)
+        url = reverse("builds_detail", args=[self.project.slug, self.build.pk])
+        resp = self.client.post(url)
+        self.assertEqual(resp.status_code, 302)
+        app.control.revoke.assert_called_once_with(
+            self.build.task_id, signal=mock.ANY, terminate=True
+        )
+        self.build.refresh_from_db()
+        self.assertEqual(self.build.state, BUILD_STATE_INSTALLING)
```
Comment (agjohnson) [inline readthedocs/builds/tests/test_views.py:42]: Why is this still `installing` at this point?

**Review by agjohnson — APPROVED** [review]
Tests look great as well :+1:

File: readthedocs/builds/tests/test_views.py (hunk 1 above)
Comment (stsewd) [reply readthedocs/builds/tests/test_views.py:42]: Changing the state of the build object will be handled inside the task

https://github.com/readthedocs/readthedocs.org/blob/4983fce383ce4b859f1c8636f139a657cfc03c7d/readthedocs/projects/tasks/builds.py#L274-L276

**Pull request merged**